src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -1142,7 +1145,10 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             .get(self, id)
             .unwrap_or(Lazy::empty())
             .decode(self)
-            .map(|index| respan(self.get_span(index, sess), self.item_name(index)))
+            .map(|index| {
+                let Ident { name, span } = self.item_name(index, sess);
+                respan(span, name)
+            })
```

Comment:
The old code was correct, we need the whole field span here, not the ident span.